### PR TITLE
Add Tests and refactors to external accounts sync api

### PR DESCRIPTION
**Pull request by Aryex82**
Issue: https://github.com/Real-Dev-Squad/website-backend/issues/1448
Parent PR:  https://github.com/Real-Dev-Squad/website-backend/pull/1468

Description: 
This is mainly a refactor PR to an existing API. 

- Adds jsdoc to users fetch functions
- Adds tests to users fetch functions
- Changes the route to POST /external-accounts/users?action=discord-users-sync
- Adds validations to query params

Test Coverage: 
Screenshot 2023-09-02 at 4 26 19 PM
Screenshot 2023-09-02 at 6 10 15 PM
Screenshot 2023-09-02 at 9 31 00 PM

Screenshot 2023-09-05 at 8 25 46 PM

### CODE REVIEW

**Comment by Pratiyushkumar** [comment]
can you add postman output by sending a request and getting a response for some visual understanding of the feature you are building.

**Review by satyam73 — CHANGES_REQUESTED** [review]

File: models/users.js [hunk 1]
```diff
@@ -750,6 +750,10 @@ const generateUniqueUsername = async (firstname, lastname) => {
   }
 };
 
+/**
+ * Updates given list of users in batch
+ * @param usersData {Array} - Users list as an array. Each element of user list should contain a property called id(PK)
```
[the remark this answers is not in the document]
Comment (Aryex82) [reply models/users.js:755]: Primary Key.

[the remark this answers is not in the document]
Comment (Aryex82) [reply models/users.js]: I've updated the doc.

**Comment by sahsisunny** [comment]
Can you also mention the PR for updating the doc like API contracts?

**Review by satyam73 — APPROVED** [review]
LGTM

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**